### PR TITLE
Code grooming - Improve utests code coverage

### PR DESCRIPTION
**Pull request by SpacePossum**
+ bit of SCA and typos fixes

### CODE REVIEW

File: Symfony/CS/Fixer/PSR2/LowercaseConstantsFixer.php [hunk 1]
```diff
@@ -81,10 +81,6 @@ private function isNeighbourAccepted(Tokens $tokens, $index)
             }
         }
 
-        if (null === $index) {
```
Comment (SpacePossum) [inline Symfony/CS/Fixer/PSR2/LowercaseConstantsFixer.php]: is never true so can be removed

File: Symfony/CS/Fixer/Symfony/TernarySpacesFixer.php [hunk 2]
```diff
@@ -81,7 +81,7 @@ public function getDescription()
         return 'Standardize spaces around ternary operator.';
     }
 
-    private function ensureWhitespaceExistance(Tokens $tokens, $index, $after)
+    private function ensureWhitespaceExistence(Tokens $tokens, $index, $after)
```
Comment (SpacePossum) [inline Symfony/CS/Fixer/Symfony/TernarySpacesFixer.php]: just a little typo ;)

File: Symfony/CS/Tests/IntegrationTest.php [hunk 3]
```diff
@@ -33,6 +33,6 @@ protected static function getFixturesDir()
      */
     protected static function getTempFile()
     {
-        return self::getFixturesDir().'/tmp/tmp.php';
+        return static::getFixturesDir().'/tmp/tmp.php';
```
Comment (SpacePossum) [inline Symfony/CS/Tests/IntegrationTest.php]: use late static binding so the method can be overridden

Comment (gharlan) [inline Symfony/CS/Tests/IntegrationTest.php]: You want to overwrite a final class? :)

Comment (SpacePossum) [inline Symfony/CS/Tests/IntegrationTest.php]: oh, ehmm no :)

I meant, it should call its own implementation of `getFixturesDir` and not the parent one (`AbstractIntegrationTestCase`).

**Comment by SpacePossum** [comment]
fabbot.io fails because it tries to fix the tests cases by itself...

File: Symfony/CS/Tests/IntegrationTest.php (hunk 3 above)
Comment (keradus) [inline Symfony/CS/Tests/IntegrationTest.php]: be we do not call it on parent-level, dont wee?

**Comment by keradus** [comment]
@fabpot we have already released stable version of PHPUnit fixers.
Please, could you update fabbot.io to use them instead of regex ?

File: Symfony/CS/Tests/ApplicationTest.php [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Symfony\CS\Tests;
+
+use Symfony\CS\Console\Application;
+
+/**
+ * @author SpacePossum
```
Comment (GrahamCampbell) [inline Symfony/CS/Tests/ApplicationTest.php]: invalid author tag

Comment (GrahamCampbell) [inline Symfony/CS/Tests/ApplicationTest.php]: Should be like this: https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/c784f68e77052939f594044f8d30e7ce1b564543/Symfony/CS/Tests/IntegrationTest.php#L17.

Comment (keradus) [inline Symfony/CS/Tests/ApplicationTest.php]: [it is valid](https://github.com/phpDocumentor/fig-standards/blob/master/proposed/phpdoc.md#examples-1)

Comment (GrahamCampbell) [inline Symfony/CS/Tests/ApplicationTest.php]: ok then, it's inconsistent with tags from this author already in the codebase

File: Symfony/CS/Tests/IntegrationTest.php (hunk 3 above)
Comment (SpacePossum) [inline Symfony/CS/Tests/IntegrationTest.php]: indeed this was not needed, reverted (this one was needed https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/1585/files#diff-f387a0305e98fb7de24f5f8699ec1ec6L92)

File: Symfony/CS/Tests/ApplicationTest.php (hunk 4 above)
Comment (keradus) [inline Symfony/CS/Tests/ApplicationTest.php]: that's true ;)

File: Symfony/CS/Tests/ApplicationTest.php [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+ */
+class ApplicationTest extends \PHPUnit_Framework_TestCase
+{
+    public function testApplication()
+    {
+        $app = new Application();
+        $this->assertStringMatchesFormat('<info>PHP CS Fixer</info> version <comment>1.12-DEV</comment> by <comment>Fabien Potencier</comment>', $app->getLongVersion());
```
Comment (stof) [inline Symfony/CS/Tests/ApplicationTest.php]: this test is very fragile, as it must be updated for any release

File: Symfony/CS/Tests/DocBlock/AnnotationTest.php [hunk 6]
```diff
@@ -238,4 +238,14 @@ public function testSetTypesOnBadTag()
 
         $tag->setTypes(array('string'));
     }
+
+    public function testGetTagsWithTypes()
+    {
+        $tags = Annotation::getTagsWithTypes();
+        $this->assertTrue(is_array($tags));
+        foreach ($tags as $tag) {
+            $this->assertTrue(is_string($tag));
```
Comment (stof) [inline Symfony/CS/Tests/DocBlock/AnnotationTest.php]: Use ``assertInternalType``

Comment (SpacePossum) [inline Symfony/CS/Tests/DocBlock/AnnotationTest.php]: thanks :)
updated

File: Symfony/CS/Tests/ToolInfoTest.php [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+class ToolInfoTest extends \PHPUnit_Framework_TestCase
+{
+    public function testGetVersion()
+    {
+        $this->assertInternalType('string', ToolInfo::getVersion());
+    }
+
+    public function testIsInstallAsPhar()
+    {
+        $this->assertFalse(ToolInfo::isInstalledAsPhar());
```
Comment (keradus) [inline Symfony/CS/Tests/ToolInfoTest.php]: does Composer could be checked ?

Comment (SpacePossum) [inline Symfony/CS/Tests/ToolInfoTest.php]: not without faking all the logic that is the same as being tested, so it would not be a very useful test  :(

Comment (keradus) [inline Symfony/CS/Tests/ToolInfoTest.php]: :(

**Comment by keradus** [comment]
Thank you @SpacePossum.

**Pull request closed without merging**